src/Microsoft.AspNet.Razor/Text/SourceLocation.cs

```diff
@@ -15,34 +15,21 @@ public struct SourceLocation : IEquatable<SourceLocation>, IComparable<SourceLoc
         public static readonly SourceLocation Undefined = CreateUndefined();
         public static readonly SourceLocation Zero = new SourceLocation(0, 0, 0);
 
-        private int _absoluteIndex;
-        private int _lineIndex;
-        private int _characterIndex;
-
         public SourceLocation(int absoluteIndex, int lineIndex, int characterIndex)
         {
-            _absoluteIndex = absoluteIndex;
-            _lineIndex = lineIndex;
-            _characterIndex = characterIndex;
+            AbsoluteIndex = absoluteIndex;
+            LineIndex = lineIndex;
+            CharacterIndex = characterIndex;
         }
 
-        public int AbsoluteIndex
-        {
-            get { return _absoluteIndex; }
-        }
+        public int AbsoluteIndex { get; set; }
```

Comment:
Ya, it's purely so it can be serialized. I guess the other approach would be to provide a custom Json.net serializer for this class on the DTH side.